crates/swc_plugin_runner/src/lib.rs

```diff
@@ -96,9 +97,37 @@ fn load_plugin(plugin_path: &Path, cache: &mut Option<PluginCache>) -> Result<In
     return match module {
         Ok(module) => {
             let memory = Memory::new(&wasmer_store, MemoryType::new(1, None, false))?;
-            let import_object = imports! {
-                "env" => {
-                    "memory" => memory
+            let wasi_version = get_wasi_version(&module, true);
```

Comment:
If plugin binary have correct & strict wasi version, we expect it is compiled as `wasm32-wasi`, setting up runtime import object for wasi. Otherwise we'll fall back to default import_object.

Reply:
I think this is a good change, but if I failed to use `wasi` with wasm compiled to `unknown-unknown`.
Seems like you find a way :)